contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -154,8 +157,23 @@ contract BosonVoucher is IBosonVoucher, BeaconClientBase, OwnableUpgradeable, ER
         uint256 _start,
         uint256 _length
     ) external onlyRole(PROTOCOL) {
-        // Make sure range start is valid
-        require(_start > 0, INVALID_RANGE_START);
+        // Prevent reservation of an empty range
+        require(_length > 0, INVALID_RANGE_LENGTH);
+
+        // Prevent overflow in issueVoucher and preMint
+        require(_start <= type(uint256).max - _length, INVALID_RANGE_LENGTH);
+
+        // Make sure that ranges are in ascending order
+        if (_rangeOfferIds.length > 0) {
+            // Get latest registered range
+            Range storage lastRange = _rangeByOfferId[_rangeOfferIds[_rangeOfferIds.length - 1]];
```

Comment:
```suggestion
        uint256[] rangeOfferIds = _rangeOfferIds;
        if (rangeOfferIds.length > 0) {
            // Get latest registered range
            Range storage lastRange = _rangeByOfferId[rangeOfferIds[rangeOfferIds.length - 1]];
```
gas optimization

Reply:
I assume you meant "uint256[] **memory** rangeOfferIds = _rangeOfferIds;" This is not an optimization, since it copies all entries from storage to memory. Already when there are two elemens in the array, that becomes more expensive than current implementation. Whit multiple ranges it gets worse.

Instead I only stored the length to variable and thus save a bit of gas.